### PR TITLE
Add better condition syncing for v2 provisioning objects

### PR DESCRIPTION
**Pull request by thedadams**
Issue: https://github.com/rancher/rancher/issues/36993

## Problem
There were issues with different pages of the UI showing different status for
the cluster based on which object it was showing (management vs. provisioning).
This change adds better syncing between the cluster objects so they have the
same conditions and statuses.

## Solution
In addition, the planner always used the Provisioned condition when
changing/applying the node plans. This was confusing, especially when a machine
was being updated. This change will use the Reconciled condition instead.

## Testing
Provisioning and upgrading clusters of all types. Making sure that the Home page and Cluster Management page.

### CODE REVIEW

File: pkg/provisioningv2/rke2/planner/planner.go [hunk 1]
```diff
@@ -173,29 +173,34 @@ func (p *Planner) applyToMachineCondition(clusterPlan *plan.Plan, machineNames [
...
-				continue
-			}
+			waiting = true
+			continue
+		}
+
+		// Once a machine is provisioned (i.e. the first time it is bootstrapped and joined to the cluster),
+		// the machine should not go into a provisioning state again. Instead, updating is used.
+		cond := rke2.Provisioned
+		if cond.IsTrue(machine) {
```
Comment (Oats87) [inline pkg/provisioningv2/rke2/planner/planner.go:183]: I'm not a fan of changing the conditions we're setting based on other conditions.

**Review by Oats87 — CHANGES_REQUESTED** [review]

File: pkg/provisioningv2/rke2/planner/planner.go (hunk 1 above)
Comment (thedadams) [reply pkg/provisioningv2/rke2/planner/planner.go:183]: @Oats87 Do you have another suggestion?

Comment (Oats87) [reply pkg/provisioningv2/rke2/planner/planner.go:183]: If the goal is to fix a superficial issue where the user may see "provisioning" on a machine in the UI, then I would lean to fixing this at the UI level. 

My understanding is that the requirement here is entirely to solve a superficial issue where a machine says "provisioning". If that is the case, then maybe we just change the condition entirely to `Reconciled` or something along those lines. 

I see no benefit of this change to the backend other than introducing potential bugs/confusion as we copy/trickle conditions around all over the place.

**Comment by thedadams** [comment]
Note to reviewers: an update to this PR will be needed after [this](https://github.com/rancher/wrangler/pull/201) PR is merged.

File: pkg/provisioningv2/rke2/planner/planner.go (hunk 1 above)
Comment (thedadams) [reply pkg/provisioningv2/rke2/planner/planner.go]: Using Reconciled instead.

**Review by paynejacob — APPROVED** [review]

**Review by Oats87 — APPROVED** [review]

**Review by Oats87 — APPROVED** [review]

**Pull request merged**